docs/BEST_PRACTICES.md

```diff
@@ -1,3 +1,69 @@
+- [Order/Cart related best practices](#ordercart-related-best-practices)
+- [Deployment best practices](#deployment-best-practices)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Order creation best practices
+#### AFTER a successful payment
+##### Limitations & Solutions
+- It is possible that shop was not reachable for network issues. In that case there will be a successful payment but no order.
+
+  ###### Possible solution:
+  Create an order based on transaction state changes (commercetools messages or subscriptions)
+
+------  
+- It is possible to have more than 1 valid payments on cart/order. During checkout in two tabs, two redirect payments
+(like credit card and paypal) can be created for the same cart. These two payments can be completed independently in both tabs.
+ This will lead the cart that links to two successful payments.
+  ###### Possible solution:
+  Refund the payment. (It can be based on commercetools messages or subscriptions)
+
+------  
+- It is possible to have the payment amount is lower than cart amount. During the checkout the shopper opens two tabs, one tab is for cart and
+the other tab is for payment provider. Shopper can add more items to the cart so that the cart value is not equal to the 
+amount that will be paid in the payment provider tab.
+
+  ###### Possible solution:
+  After a successful payment, validate the cart to check if the cart amount matches payment amount. 
+  If not, refund the payment and let shopper to pay again.
+
+#### BEFORE a successful payment
+##### Limitations
+- If the shopper never finishes (according to the stats most shoppers jump out of checkout when they actually have to pay) or cancels a payment the reserved stock with an order will not be released and thus there is a risk that simple automation (or even many users) will run the shop out of stock
+------  
+- There will be orders which will never be paid but shown in commercetools Merchant Center dashboard as revenue - which is not true
+------  
+- Not paid orders will falsely boost the concept of “best sellers” index of its products.
+------  
+- Creating an order does not allow you to modify the original cart anymore. What happens if a shopper forgot something and would like to change the cart just after initiation of the payment?
+------  
+- How to handle vouchers that can be applied only once but the shopper decided not to finish the payment and change the cart instead? It might require a new cart based on the old cart. The creation of a new cart with the same item might be an issue if an item had only stock = 1 and it has been used in not paid order already.
+
+##### Conclusion
+Both approaches have their good and bad sides, but it is recommended creating order AFTER a successful payment is less harmful.
+
+## Payment related best practices
+
+- Payment status changes will not be managed by the Adyen Integration library. Adyen Integration is base upon the 
```

Comment:
Maybe better:
`Currently [payment status](https://docs.commercetools.com/api/projects/payments#paymentstatus) is not maintained by the integration as status of the payment can be derived from the state of the payment transaction(s).`

Reply:
198229a6c481d22ca386cfdc44990d0032ba1939